src/hooks/useTokenBalance.ts

```diff
@@ -0,0 +1,45 @@
+import { useContext, useEffect, useState } from "react";
+import BN from "bn.js";
+
+import { Web3Context } from "components/Web3Provider";
+
+export interface UseTokenBalanceResult {
+  balance: BN | null;
+  isLoading: boolean;
+}
+
+/**
+ * Fetches balance of current Safe for given token address.
+ * Returns the balance as BN, if any
+ *
+ * @param tokenAddress Address of token to query the balance for current Safe
+ */
+export function useTokenBalance(tokenAddress?: string): UseTokenBalanceResult {
+  const [balance, setBalance] = useState<BN | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const { safeInfo, getContract } = useContext(Web3Context);
+
+  // TODO: cache responses so we don't have to query the network on every single change
+  // TODO: also, we should subscribe to new block header if available and query the balance
+  // on every change
+  useEffect(() => {
+    setIsLoading(true);
```

Comment:
Could the below code fail? If so, it makes sense to alter this to have a status for `ERROR` aswell.

Reply:
https://github.com/gnosis/safe-cmm-app-react/pull/24/commits/a1f7bc2a1fb6ba4cfa9a039f9db13a3528fbe7b7